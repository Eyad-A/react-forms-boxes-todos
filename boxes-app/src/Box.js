import React from "react";

function Box({
    id,
    handleRemove,
    width=200,
    height=200,
    backgroundColor = "green"
}) {
    const remove = () => handleRemove(id);
    return (
        <div>
            <div 
                style={{
                    height: `${height}px`,
                    width: `${width}px`,
                    backgroundColor 
                }}
            />
            <button onClick={remove}>X</button>            
        </div>
    );
}

export default Box;