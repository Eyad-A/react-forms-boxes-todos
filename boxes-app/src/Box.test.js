import React from "react";
import { render } from '@testing-library/react';
import Box from './Box';

it('should render the Box component', () => {
    render(<Box />);
});

it('matches the snapshot', function() {
    const {asFragment} = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});