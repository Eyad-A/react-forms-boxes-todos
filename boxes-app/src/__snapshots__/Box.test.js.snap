// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="height: 200px; width: 200px; background-color: green;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
